nzedb/NameFixer.php

```diff
@@ -1494,4 +1578,38 @@ public function fileCheck($release, $echo, $type, $namestatus, $show)
 			}
 		}
 	}
+
+	/**
+	 * Look for a name based on mediainfo xml Unique_ID.
+	 *
+	 * @param         $release
+	 * @param         $release
+	 * @param boolean $echo
+	 * @param string  $type
+	 * @param         $namestatus
+	 * @param         $show
+	 */
+	public function uidCheck($release, $echo, $type, $namestatus, $show)
+	{
+		if ($this->done === false && $this->relid !== $release["releases_id"]) {
+			$result = $this->pdo->queryExec(
```

Comment:
You are selecting almost every release that has a uniqueid every single iteration.  Point the query against the release_uniqueid table instead and join releases to it once you've found a Pre match by the uid like this:

```php
			$result = $this->pdo->queryOneRow("
						SELECT r.id AS releases_id, r.size AS relsize, r.name AS textstring, r.searchname AS searchname, r.predb_id, HEX(ru.uniqueid) AS uid
						FROM release_unique ru ON ru.releases_id = r.id
						LEFT JOIN release_unique ru ON (ru.releases_id = r.id)
						WHERE r.predb_id > 0 AND ru.uniqueid = UNHEX('{$release['uid']}')"
					);
```

This way we are only looking up and retrieving a single row.